samza-core/src/main/java/org/apache/samza/storage/TaskStorageCommitManager.java

```diff
@@ -139,18 +132,16 @@ public void init() {
   }
 
   /**
-   * // TODO HIGH dchen fix method name and documentation. local file system isn't very relevant here (too general).
-   * // Maybe rename to "writeCheckpointToStoreDirs" or something and update docs.
    * Writes the {@link Checkpoint} returned by {@link #commit(CheckpointId)}
    * locally to the file system on disk if the checkpoint passed in is an instance of {@link CheckpointV2},
    * otherwise if it is an instance of {@link CheckpointV1} persists the Kafka changelog ssp-offsets only.
    *
    * Note: The assumption is that this method will be invoked once for each {@link Checkpoint} version that the
-   * task needs to write.
+   * task needs to write, once per checkpoint version for backwards compatibility.
    *
    * @param checkpoint the latest checkpoint to be persisted to local file system
    */
-  public void persistToLocalFileSystem(Checkpoint checkpoint) {
+  public void writeCheckpointToStoreDirectory(Checkpoint checkpoint) {
```

Comment:
`Directories` (since this writes to both store dir and store checkpoint dir). Can also clarify this in the method javadocs. E.g.: 

"Writes the {@link Checkpoint} information returned by {@link #commit(CheckpointId)} in each store directory and store checkpoint directory. For CheckpointV2, writes the entire task {@link CheckpointV2}. For CheckpointV1, only writes the changelog ssp offsets in the OFFSET* files."